nemo/collections/tts/parts/utils/callbacks.py

```diff
@@ -229,6 +229,39 @@ def on_train_epoch_end(self, trainer: Trainer, model: LightningModule):
             self._log_image(image=image, log_dir=log_dir, step=model.global_step)
 
 
+class VocoderArtifactGenerator(ArtifactGenerator):
+    """
+    Generator for logging Vocoder model outputs.
+    """
+
+    def generate_artifacts(
+        self, model: LightningModule, batch_dict: Dict
+    ) -> Tuple[List[AudioArtifact], List[ImageArtifact]]:
+
+        audio_artifacts = []
+
+        audio_filepaths = batch_dict.get("audio_filepaths")
+        audio_ids = [create_id(p) for p in audio_filepaths]
+
+        audio = batch_dict.get("audio")
+        audio_len = batch_dict.get("audio_lens")
+
+        spec, spec_len = model.audio_to_melspec_precessor(audio, audio_len)
+
+        with torch.no_grad():
+            # [B, T]
+            audio_pred = model.forward(spec=spec).squeeze(1)
```

Comment:
(A nitpick)
Would be a nice place to use `einops.rearrange`, it is in NeMo dependencies. Makes tensor dimension dances **much** more explicit and readable.
```
audio_pred = rearrange(audio_pred, "batch 1 time -> batch time")
```

Reply:
Changed